konveyor/pelorus
# Rewrite pre-commit, bump_version, add more to dev env setup

This will make setting up an environment easier for new users, and make some of the tooling clearer to use and maintain.

## Testing Instructions

1. Remove your venv and git pre-commit hook and run `make dev-env`.
2. Activate your virtual environment if necessary. Your editor might do this for you if you open a new terminal.
3. Make a change to python code that isn't in line with the black style, or isort style. Deleting blank lines or inserting them between functions/classes is usually enough. Also try making a change that would fail pylava linting, like making a line over 120 characters.
4. Make a change to a helm chart that would necessitate a version bump.
5. Try to commit-- you should see your commit be rejected.
6. Run `make format`. You should see the python formatting be fixed.
7. Run `./scripts/chart-check-and-bump`. You should see the chart's version increase.
8. Try to commit again-- it should succeed.

@redhat-cop/mdt
